api/app/models/instance.rb

```diff
@@ -9,25 +9,17 @@ class Instance < ApplicationRecord
   has_many :conditions, dependent: :destroy
 
   scope :managements, -> { joins(:node).includes(:conditions).where('nodes.type = ?', 'HealthCares::Management') }
-  scope :variables, lambda {
-                      joins(:node).includes(:conditions).where('nodes.type IN (?)', Variable.descendants.map(&:name))
-                    }
-  scope :questions_sequences, lambda {
-                                joins(:node).includes(:conditions).where('nodes.type IN (?)', QuestionsSequence.descendants.map(&:name))
-                              }
+  scope :variables, lambda { joins(:node).includes(:conditions).where('nodes.type IN (?)', Variable.descendants.map(&:name)) }
+  scope :questions_sequences, lambda { joins(:node).includes(:conditions).where('nodes.type IN (?)', QuestionsSequence.descendants.map(&:name)) }
   scope :drugs, -> { joins(:node).includes(:conditions).where('nodes.type = ?', 'HealthCares::Drug') }
   scope :diagnoses, -> { joins(:node).includes(:conditions).where('nodes.type = ?', 'Diagnosis') }
 
-  scope :triage_complaint_category, lambda {
-                                      joins(:node).where('nodes.stage = ? AND nodes.type = ?', Variable.stages[:triage], 'Variables::ComplaintCategory')
-                                    }
-  scope :triage_under_complaint_category, lambda {
-                                            joins(:node).where('nodes.type NOT IN (?)', %w[Variables::UniqueTriageQuestion Variables::ComplaintCategory])
-                                          }
+  scope :triage_complaint_category, lambda { joins(:node).where('nodes.stage = ? AND nodes.type = ?', Variable.stages[:triage], 'Variables::ComplaintCategory') }
+  scope :triage_under_complaint_category, lambda { joins(:node).where('nodes.type NOT IN (?)', %w[Variables::UniqueTriageQuestion Variables::ComplaintCategory]) }
 
   # Allow to filter if the node is used as a health care condition or as a final diagnosis condition. A node can be used in both of them.
-  scope :diagnosis_conditions, -> { includes(:conditions).where(diagnosis_id: nil) }
-  scope :health_care_conditions, -> { includes(:conditions).where.not(diagnosis_id: nil) }
+  scope :decision_tree_diagram, -> { includes(:conditions).where(diagnosis_id: nil) }
+  scope :diagnoses_diagram, -> { includes(:conditions).where.not(diagnosis_id: nil) }
```

Comment:
This scope is not used

Reply:
En effet, j'utilise le has_many components dans diagnosis maintenant, qui est plus propre